astropy/coordinates/tests/test_frames.py

```diff
@@ -711,3 +711,40 @@ def test_component_error_useful():
         i.lon  # lon is *not* the component name despite being the underlying representation's name
     assert "object has no attribute 'foobar'" in str(excinfo1.value)
     assert "object has no attribute 'lon'" in str(excinfo2.value)
+
+
+def test_cache_clear():
+    from ..builtin_frames import ICRS
+
+    i = ICRS(1*u.deg, 2*u.deg)
+
+    # Add an in frame units version of the rep to the cache.
+    repr(i)
+
+    assert len(i.cache['representation']) == 2
+
+    del i.cache['representation']
```

Comment:
Given what I just stated for the documentation, I think this should be `i.cache.clear()` instead.